examples-2022/calc/src/type_check.rs

```diff
@@ -58,11 +58,24 @@ impl CheckExpression<'_> {
                 }
             }
             crate::ir::ExpressionData::Call(f, args) => {
-                if self.find_function(*f).is_none() {
-                    self.report_error(
+                match self.find_function(*f) {
+                    Some(func) => {
+                        let expected_arity = func.args(self.db).len();
+                        let actual_arity = args.len();
+                        if actual_arity != expected_arity {
+                            self.report_error(
+                                expression.span,
+                                format!(
+                                    "the function `{}` expects {expected_arity} argument(s), only {actual_arity} provided",
```

Comment:
```suggestion
                                    "the function `{}` expects {expected_arity} argument(s), but {actual_arity} provided",
```
I think it would be better to change it to "but", because the `actual_arity` may be more than the `expected_arity`, as in your test sample below